dll/win32/shell32/CShellLink.cpp

```diff
@@ -2592,40 +2592,33 @@ HRESULT STDMETHODCALLTYPE CShellLink::InvokeCommand(LPCMINVOKECOMMANDINFO lpici)
 
 HRESULT CShellLink::DoOpen(LPCMINVOKECOMMANDINFO lpici)
 {
-    HRESULT hr;
-    LPWSTR args = NULL;
-    LPWSTR path = strdupW(m_sPath);
     BOOL unicode = lpici->cbSize >= FIELD_OFFSET(CMINVOKECOMMANDINFOEX, ptInvoke) &&
                    (lpici->fMask & CMIC_MASK_UNICODE);
 
+    CStringW args;
+    if (m_sArgs)
+        args = m_sArgs;
+
+    CComHeapPtr<WCHAR> pszParams;
```

Comment:
I would suggest you move this to where it is used (just before the `__SHCloneStrAtoW(&pszParams, lpici->lpParameters))` call below).
So that we don't keep a copy in memory even after it's not needed anymore. When it's concatenated into `args`, the string has been copied to the args, and we should be able to free `pszParams`.

Reply:
Moved.